ocp_resources/resource.py

```diff
@@ -446,16 +450,53 @@ def to_dict(self):
 
     def __enter__(self):
         signal(SIGINT, self._sigint_handler)
-        return self.deploy()
+        """
+        For debug, export REUSE_IF_RESOURCE_EXISTS to skip resource create.
+        Examples:
+            To skip creation of all resources by kind:
+                export REUSE_IF_RESOURCE_EXISTS="{Pod: {}}"
+
+            To skip creation of resource by name (on all namespaces):
+                export REUSE_IF_RESOURCE_EXISTS="{Pod: {<pod-name>: ''}}"
+
+            To skip creation of resource by name and namespace:
+                export REUSE_IF_RESOURCE_EXISTS="{Pod: {<pod-name>: <pod-namespace>}}"
+        """
+        _resource = None
+        _export_str = "REUSE_IF_RESOURCE_EXISTS"
+        skip_resource_kind_create_if_exists = os.environ.get(_export_str)
+        if skip_resource_kind_create_if_exists:
+            _resource = skip_existing_resource_creation_teardown(
+                resource=self,
+                export_str=_export_str,
+                user_exported_args=skip_resource_kind_create_if_exists,
+            )
+
+        return _resource or self.deploy()
 
     def __exit__(self, exception_type, exception_value, traceback):
-        # For debug, export kind (class name) + NOTEARDOWN to skip resource teardown.
-        # For example:export VirtualMachineInstanceNOTEARDOWN=True
-        skip_teardown_env = f"{self.kind}NOTEARDOWN"
-        no_teardown_from_environment = os.environ.get(skip_teardown_env)
-        if no_teardown_from_environment:
+        """
+        For debug, export SKIP_RESOURCE_TEARDOWN to skip resource create.
```

Comment:
```suggestion
        For debug, export SKIP_RESOURCE_TEARDOWN to skip resource teardown.
```

relevant in the comment below as well